src/libraries/System.Console/src/System/ConsolePal.Windows.cs

```diff
@@ -122,10 +124,25 @@ public static void SetConsoleOutputEncoding(Encoding enc)
             if (enc.CodePage != UnicodeCodePage)
             {
                 if (!Interop.Kernel32.SetConsoleOutputCP(enc.CodePage))
-                    throw Win32Marshal.GetExceptionForWin32Error(Marshal.GetLastPInvokeError());
+                {
+                    HandleSetConsoleEncodingError();
+                }
             }
         }
 
+        private static void HandleSetConsoleEncodingError()
+        {
+            int lastError = Marshal.GetLastPInvokeError();
```

Comment:
It may be better to pass the last error into this method. It would be easy for future refactoring to adda line of code before the call to this method that resets last error. Better to store it immediately after it is set.